test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -0,0 +1,591 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "ucx_wrapper.h"
+
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <string.h>
+#include <assert.h>
+
+#include <iostream>
+#include <limits>
+#include <sstream>
+
+
+struct ucx_request {
+    UcxCallback  *callback;
+    bool         completed;
+    uint32_t     conn_id;
+    ucs_status_t status;
+};
+
+class debug {
+public:
+    template<typename T>
+    debug& operator<<(const T &t) {
+#if _DEBUG
```

Comment:
can we have runtime check if debug is enabled instead of compile time? should not affect performance too much and would be helpful in default release build

Reply:
rename to `verbose` and enable according to `test_opts->verbose`?

Reply:
BTW. runtime check won't remove cascade of `operator <<` calls

Reply:
rename to verbose - yes
let's make verbose() ass simple as possible, one function like log()